tests/unit/modules/test_roles.py

```diff
@@ -0,0 +1,304 @@
+import json
+from unittest.mock import Mock
+from unittest.mock import patch
+
+import pytest
+import saltext.vmware.modules.roles as security_roles
+import saltext.vmware.utils.drift as drift
+from pyVmomi import vim
+
+
+def mock_with_name(name, *args, **kwargs):
+    # Can't mock name via constructor: https://docs.python.org/3/library/unittest.mock.html#mock-names-and-the-name-attribute
+    mock = Mock(*args, **kwargs)
+    mock.name = name
+    return mock
+
+
+def mock_pyvmomi_role_object(roleId, name, label, new_privileges=[]):
+    """
+
+    .. code-block:: json
+    (vim.AuthorizationManager.Role) {
```

Comment:
**suggestion** need a blank line and then indention x4 spaces for code blocks.

Reply:
(the only reason this didn't cause an error is that docstrings for things in tests aren't built as part of our test runs)

Reply:
I rely on docstrings to catch it, I'll pay more attention about this in the tests
done for both tests